packages/common/src/visualizations/CartesianChartDataTransformer.ts

```diff
@@ -1,39 +1,26 @@
 import { friendlyName } from '../types/field';
 import { ChartKind } from '../types/savedCharts';
-import {
-    isCartesianChartSQLConfig,
-    type SqlRunnerChartConfig,
-} from '../types/sqlRunner';
 import { type ResultsRunnerBase } from './ResultsRunnerBase';
 import { type IndexType } from './SqlResultsRunner';
 
-export class CartesianChartDataTransformer<
-    TCartesianChartLayout,
-    TPieChartConfig,
-> {
-    private readonly transformer: ResultsRunnerBase<
-        TCartesianChartLayout,
-        TPieChartConfig
-    >;
+export class CartesianChartDataTransformer<TPivotChartLayout> {
+    private readonly transformer: ResultsRunnerBase<TPivotChartLayout>;
 
-    constructor(args: {
-        transformer: ResultsRunnerBase<TCartesianChartLayout, TPieChartConfig>;
-    }) {
+    constructor(args: { transformer: ResultsRunnerBase<TPivotChartLayout> }) {
         this.transformer = args.transformer;
     }
 
-    async getEchartsSpec(config: SqlRunnerChartConfig) {
-        if (!isCartesianChartSQLConfig(config)) {
+    async getEchartsSpec(
```

Comment:
Improve typing by exploding the config into `layout` (not used by chart models only runner) and `display` (not used by runner only chart models). i.e. `layout` is opaque to chart (since it depends on the runner context)